### PR TITLE
Methods for: attenuation in volum material, and size of the sun

### PR DESCRIPTION
**Pull request by RamonPujol**

### CODE REVIEW

File: raytrace.py [hunk 1]
```diff
@@ -185,13 +193,14 @@ def change_of_direction(self, ray, normal_vector):
 
 
 class VolumeMaterial(Material, object):
-    def __init__(self, name, properties):
+    def __init__(self, name, index_of_refraction, extinction_coefficient = None):
```
Comment (bielcardona) [inline raytrace.py]: I would keep the old signature, more generic than the new one. This way, if you need to introduce new parameters, you don't have to redefine the __init__ method.

**Review by bielcardona — COMMENTED** [review]
See comment in code

File: raytrace.py (hunk 1 above)
Comment (RamonPujol) [reply raytrace.py]: I agree. I will make the changes in a new pull request.

File: raytrace.py [hunk 2]
```diff
@@ -581,9 +726,16 @@ def next_direction(self, face):
...
         # TODO: @Ramon
-        pass
+        solid_1 = self.scene.solid_at_point(point_1)
+        solid_2 = self.scene.solid_at_point(point_2)
+        if solid_1 and solid_1 == solid_2:
+            if 'extinction_coefficient' in self.scene.materials[solid_1].properties:
+                k = self.scene.materials[solid_1].properties['extinction_coefficient'](self.wavelength)
+                if k > 0:
+                    d = point_1.distanceToPoint(point_2)
+                    self.energy = self.energy * math.exp(- k * d / 1000.0)
```
Comment (bielcardona) [inline raytrace.py]: there is no need to take point_1 and point_2 as parameters. You can recover them from self.

Comment (RamonPujol) [reply raytrace.py]: OK, I will do it in the next pull request

File: raytrace.py [hunk 3]
```diff
@@ -583,7 +729,16 @@ def next_direction(self, face):
 
     def update_energy(self):
         # TODO: @Ramon
-        pass
+        point_1 = self.points[-1]
+        point_2 = self.points[-2]
+        solid_1 = self.scene.solid_at_point(point_1)
+        solid_2 = self.scene.solid_at_point(point_2)
+        if solid_1 and solid_1 == solid_2:
```
Comment (bielcardona) [inline raytrace.py]: I'd suggest just to look at the midpoint of point_1 and point_2. point_1 and point_2 are on the faces and the solid_at_point method could fail. Also, this way you only have one material to look at.

Comment (RamonPujol) [reply raytrace.py]: I completely agree. I will amend it in the next pull request.

**Pull request merged**